### ISSUE TITLE
Controlling embedded jetty logging

### ISSUE THREAD

**Issue by janikgithub** [eclipse/jetty.project#5232]
Hello, I am using embedded Jetty 9.4.15.v20190215. Logging level of my application is DEBUG. 
Every REST call places http headers into the log file. I assume it is done by the Jetty because my app. doesn't do it.
Example:
```
org.eclipse.jetty.server.HttpChannel - REQUEST for ...
POST ... HTTP/1.1
Content-Type: application/x-www-form-urlencoded
....
```

I would like to show only some of the headers.
I looked at the [Example](https://gist.github.com/joakime/27c42782c9a2712a1010), but decided to choose a more simple approach:

- Created `MyCustomRequestLog extends CustomRequestLog`
- implemented method `public void log(Request request, Response response)` and log only some of the headers
- attached the custom `RequestLog `to Jetty `server.setRequestLog(new MyCustomRequestLog());`

Although I see the log produced by my  `MyCustomRequestLog ` all headers are still placed in the log file.
Any help is appreciated.
Thanks

**Comment by lachlan-roberts** [comment]
@janikgithub I would not recommend using the `CustomRequestLog` like that. You can customize the output of the `CustomRequestLog` with the format string, and you can adjust the destination of the output logs with the the `RequestLog.Writer`. By changing the format string you can output only the request headers you want to see using the `%{VARNAME}i` code, you should not need to subclass it. 

Take a look at the [javadoc](https://www.eclipse.org/jetty/javadoc/current/org/eclipse/jetty/server/CustomRequestLog.html) for `CustomRequestLog` for more info, and also the documentation at https://www.eclipse.org/jetty/documentation/current/configuring-jetty-request-logs.html

The request log is not added by default into Jetty, you would have to configure this yourself, and it will not change the existing debug logs for jetty classes, you can change the logging level to INFO for these jetty classes if you do not wish to see it.

**Comment by janikgithub** [comment]
Thank you @lachlan-roberts. You mentioned changing the logging level to INFO for jetty classes. How do I do this? Thanks

**Comment by joakime** [comment]
When you mention DEBUG and logging you are referring to the normal JVM wide logging.

That can be done with any number of Logging libraries. (slf4j, log4j, logback, java.util.logging, commons-logging, juli, etc)

The `CustomRequestLog` is specifically for producing a "Request Log" (aka: Access Log, or NCSA Access Log).
That is limited to producing a logging event for each completed http exchange.
The `RequestLog` concept has no concept of logging levels, such as DEBUG.
The `RequestLog` layer is probably a distraction, unrelated to your core question.

Some questions for you:

1. What Logging Library are you using? (one of, or a combination of, of the following: slf4j, log4j, logback, java.util.logging, commons-logging, juli, etc)
2. When you say "Logging level of my application is DEBUG" what does that mean?
3. How do you configure that Logging Library? (what xml file, what properties file, what system properties, etc are you configuring?)
4. What does your Logging Library Configuration look like currently?

Once you can answer those questions then you can use the documentation of that Logging Library to configure specific named loggers (all logging libraries support this concept), to limit the logging level at that named logger tree.   Jetty logs to the tree starting at `org.eclipse.jetty.`, so you could just set that named logger tree to level to INFO or WARN and stop seeing DEBUG events from Jetty itself.

In short, the key takeaway is that you are not configuring Jetty, you are configuring your chosen Logging Library.

**Comment by janikgithub** [comment]
Thank you @joakime. I am using logback for logging. I found this [Embedded Jetty 9 Logging with Logback](https://stackoverflow.com/questions/26506318/embedded-jetty-9-logging-with-logback) and will try to follow it.
